### ISSUE TITLE
Uploading clock stop at 70%

### ISSUE THREAD

**Issue by arcanisgk** [phpmyadmin/phpmyadmin#18600]
SO: Ubuntu 20.04
phpMyadmin version: 5.2.1
Method installation: manual.

I don't know 100% what is expected from the functionality of the clock that is used to measure the upload percentage of the files, when it is important, but as a final user who has tried it I see something that seems strange to me:

1. They should separate the uploading of the files from the processing and dumping of the database records.
2. I don't know how phpmyadmin executes the processing and dumping of the database records in general, but for example to measure the progress in cli I usually use the pv utility with the following command:
`pv /etc/data.sql | mysql -u root -p --force --compression-algorithms=zstd`
3. Currently the upload is stuck at 70% on large files, and it's not until the dump ends that it marks it's done, I don't see it getting to 100% upload first, as shown in this image.

![image](https://github.com/phpmyadmin/phpmyadmin/assets/1207306/5c7bb5c8-be27-48d9-abea-e7828ace0d81)

**Comment by williamdes** [comment]
Hi,
Thanks for reporting this, it's not a perfect implementation. I already tried to debug this multiple times and did not find any real culprit.
The way we know the progress is a php extension: uploadprogress
See: https://github.com/phpmyadmin/phpmyadmin/blob/bfc3852e612120627edace7543167f96a3cdf8f5/libraries/classes/Plugins/Import/Upload/UploadProgress.php#L68

**Comment by arcanisgk** [comment]
> Hi, Thanks for reporting this, it's not a perfect implementation. I already tried to debug this multiple times and did not find any real culprit. The way we know the progress is a php extension: uploadprogress See:
> 
> https://github.com/phpmyadmin/phpmyadmin/blob/bfc3852e612120627edace7543167f96a3cdf8f5/libraries/classes/Plugins/Import/Upload/UploadProgress.php#L68

@williamdes  ok, I understand, so is it a bug or related to // @see https://pecl.php.net/package/uploadprogress

there is no way to test and validate that it has really reached 100% as this is reported to the ajax so that graphically it has the visual changes that are really expected...

It's getting out of hand. I don't know how to work with github and the issue of branches and commits, I haven't mastered it yet.

But then you shouldn't add other tags to this issue, like the bug has already been validated or some tag relating that the bug is in a third-party script?

**Comment by williamdes** [comment]
> @williamdes ok, I understand, so is it a bug or related to // @see https://pecl.php.net/package/uploadprogress

Most probably yes, but maybe also/only in our codebase

> there is no way to test and validate that it has really reached 100% as this is reported to the ajax so that graphically it has the visual changes that are really expected...

Technically you could do a sample upload form on your end without phpmyadmin and test that it works correctly

> It's getting out of hand. I don't know how to work with github and the issue of branches and commits, I haven't mastered it yet.

I could help you if you want, for fixes to reach 5.2 please always use QA_5_2 as a base branch for your branches. 

> But then you shouldn't add other tags to this issue, like the bug has already been validated or some tag relating that the bug is in a third-party script?

We have a label "waiting-on-upstream" but for now nothing confirms that it's in a third party
nor that it's a phpMyAdmin bug, that's why the confirmed/x.y labels are missing on this issue 
I definitely need to check that on a new Ubuntu VM with what you reported in the other issue

**Comment by arcanisgk** [comment]
> > @williamdes ok, I understand, so is it a bug or related to // @see https://pecl.php.net/package/uploadprogress ...

@williamdes Hello, I think this may be related to how web notifications are handled, apparently once the database dump starts on import, everything in phpmyadmin is blocked... I think we should migrate to a different technology than SPA , something that allows us to continue working at least with phpmyadmin and at the same time monitor the database dump work... and if a table or database cannot be seen because it is locked because a dump is being done, show a message related to that... but I don't see the need to block all of phpmyadmin.

**Comment by williamdes** [comment]
Yeah while doing some operations it's a pain everything is locked. My first path would be to look if php is preventing access to the session.
The issue is in the back end for sure. Or maybe it is the DB server blocking a table or the connection. But that seems less possible.
Could you help with the investigation?

**Comment by kamil-tekiela** [comment]
I am pretty sure it's the session locking everything up. We'd need a better implementation for session.

**Comment by arcanisgk** [comment]
I would like to but I don't know where to start, I don't even know the structure of phpmyadmin.

**Comment by williamdes** [comment]
Is this solved ?
